app/concerns/evernote_concern.rb

```diff
@@ -0,0 +1,52 @@
+module EvernoteConcern
+  extend ActiveSupport::Concern
+
+  included do
+    include Oauthable
+
+    validate :validate_evernote_options
+
+    valid_oauth_providers :evernote
+
+    gem_dependency_check { defined?(EvernoteOAuth) && Devise.omniauth_providers.include?(:evernote) }
+  end
+
+  def evernote_client
+    EvernoteOAuth::Client.new(
+      token:           evernote_oauth_token,
+      consumer_key:    evernote_consumer_key,
+      consumer_secret: evernote_consumer_secret,
+      sandbox:         use_sandbox?
+    )
+  end
+
+  private
+
+  def use_sandbox?
+    ENV["USE_EVERNOTE_SANDBOX"] == "true"
+  end
+
+  def validate_evernote_options
+    unless evernote_consumer_key.present? &&
+      evernote_consumer_secret.present? &&
+      evernote_oauth_token.present?
+      errors.add(:base, "Evernote consumer_key, consumer_secret, oauth_token, and oauth_token_secret are required to authenticate with the Evernote API.  You can provide these as options to this Agent, or as Credentials with the same names, but starting with 'evernote_'.")
```

Comment:
I think this message is wrong.  You're (correctly) using a service, so credentials / options aren't needed and look like they wouldn't be used.  I'd probably add a check on `evernote_oauth_token_secret.present?` and change the message to indicate that the ENV variables and a service are required.